lib/memery.rb

```diff
@@ -44,16 +44,13 @@ def define_memoized_method!(method_name, condition: nil)
             return super(*args, &block)
           end
 
-          @_memery_memoized_values ||= {}
+          key = "#{method_name}_#{mod_id}"
 
-          key = [method_name, mod_id].join("_").to_sym
-          store = @_memery_memoized_values[key] ||= {}
+          store = @_memery_memoized_values&.[](key)
 
-          if store.key?(args)
-            store[args]
-          else
-            store[args] = super(*args)
-          end
+          return store[args] if store&.key?(args)
+
+          ((@_memery_memoized_values ||= {})[key] ||= {})[args] = super(*args)
```

Comment:
This is super cryptic, I think we should keep things more readable.

Reply:
OK, I've splitted this line.